Improve documentation for testing

dsgkirkby opened the pull request:
This PR:
- Removes duplicate examples in the testing documentation
- Adds a new test to the Single Target example, testing a full drag-and-drop interaction
- Adds this same test to the testing documentation
- Adds a documentation section explaining what needs to be changed in the examples to work with Enzyme (addresses #925)

Note: I thought about adding a full-fledged test with Enzyme but I thought that adding Enzyme and an adapter as dependencies was a bit too heavyweight when a brief note in the docs should suffice. I'd be happy to update this with an Enzyme test if it's something that you think would be valuable.

darthtrevino commented:
@dsgkirkby Can you update your PR or allow us to update the source branch? There should be an option for that in the PR

dsgkirkby commented:
I have 'allow edits from maintainers' checked; is there something else I need to do?
I can also quickly fix this up tonight.

dsgkirkby commented:
I've updated the review from master.

darthtrevino commented on packages/documentation/docs/00 Quick Start/Testing.md:
```diff
@@ -158,86 +157,56 @@ import TestUtils from 'react-dom/test-utils';
-        return <DecoratedComponent {...this.props} />;
-      }
+it('can simulate a full drag and drop interaction', function() {
+  var DustbinWithBox = createReactClass({
```
const

darthtrevino replied:
Can you use a SFC here?

darthtrevino commented on packages/documentation/docs/00 Quick Start/Testing.md:
```diff
@@ -158,86 +157,56 @@ import TestUtils from 'react-dom/test-utils';
-  expect(div.style.opacity).toEqual(0.4);
-
-  // See other backend.simulate* methods for more!
+    var DustbinWithBoxContext = wrapInTestContext(DustbinWithBox);
```
Use const

darthtrevino commented on packages/documentation/docs/00 Quick Start/Testing.md:
````diff
@@ -158,86 +157,56 @@ import TestUtils from 'react-dom/test-utils';
+[Enzyme](https://github.com/airbnb/enzyme) is a commonly-used tool for testing React components.
+To use it with react-dnd, you'll need to call `.instance()` on `mount`-ed nodes to access the react-dnd helper methods:
+```js
+var root = Enzyme.mount(<DustbinWithBoxContext />);
````
const

darthtrevino commented on packages/documentation/docs/00 Quick Start/Testing.md:
```diff
@@ -158,86 +157,56 @@ import TestUtils from 'react-dom/test-utils';
-  // Render with the test context that uses the test backend
-  const BoxContext = wrapInTestContext(Box);
-  const root = TestUtils.renderIntoDocument(<BoxContext name='test' />);
+### Testing with Enzyme
```
This enzyme example is great, but I don't want to blow away the docs describing the standard tooling. Can you make these separate section instead?

dsgkirkby replied:
Maybe it's changed since the original PR, but when I first made this change the examples were duplicates of each other?

dsgkirkby commented:
I've updated the review to not clobber the old examples, as they now look like they're examples of doing the same thing with different levels of language features. I've implemented the full drag-and-drop example just for the first section in ES5, as it seems unnecessarily repetitive to add it for each section.
